### PR TITLE
Analysis/ standard components

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added a SideTabs component for displaying vertical tabs. Also, added a WarningAlert component which shows an icon by default. 

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/webapp/resources/js/components/alerts/WarningAlert.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Component to render a Warning alert with an icon
+ */
+
+import React from "react";
+import PropTypes from "prop-types";
+import { Alert } from "antd";
+
+export function WarningAlert(props) {
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/alerts/WarningAlert.jsx]: To all of these, can you use the following pattern:

```js
export function InfoAlert({ message, description, ...props }) {
  return (
    <Alert
      type="info"
      showIcon
      message={message}
      description={description}
      {...props}
    />
  );
}
```

Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/alerts/WarningAlert.jsx]: Sure!

File: src/main/webapp/resources/js/components/tabs/SideTabs.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const VerticalTabs = styled(Tabs)`
+  .ant-tabs-tab {
+    @media only screen and (min-width: 800px) {
+      width: 200px;
+    }
+  }
+`;
+
+export function SideTabs(props) {
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/tabs/SideTabs.jsx]: Here you could set the defaults in the method declaration:

```js
export function SideTabs({
  type = "card",
  tabPosition = "left",
  animated = false,
  activeKey,
  onChange,
  children,
  ...props
}) {
  return (
    <VerticalTabs
      type={type}
      activeKey={activeKey}
      onChange={onChange}
      tabPosition={tabPosition}
      animated={animated}
      {...props}
    >
      {children}
    </VerticalTabs>
  );
}
```

File: src/main/webapp/resources/js/components/tabs/SideTabs.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    </VerticalTabs>
+  );
+}
+
+SideTabs.propTypes = {
+  type: PropTypes.string,
+  activeKey: PropTypes.string,
+  onChange: PropTypes.func,
+  tabPosition: PropTypes.string,
+  animated: PropTypes.bool
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/tabs/SideTabs.jsx]: Don't forget `children` inhere.  You also should add to your docs what the props are.  Try following https://devdocs.io/jsdoc/

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx [hunk 4]
```diff
@@ -9,10 +9,11 @@
 
 import React, { Suspense, useContext, useEffect, useState } from "react";
 import { AnalysisContext } from "../../../contexts/AnalysisContext";
-import { Alert, Spin, Tabs } from "antd";
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Create a `ContentLoading` component that can accept a message (maybe "Loading..." or something).  That way if we want to update it at any point we can do them all at once.

File: src/main/webapp/resources/js/components/lists/BasicList.jsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  ...props
+}) {
+  return (
+    <List
+      itemLayout="horizontal"
+      dataSource={dataSource}
+      renderItem={item => (
+        <List.Item>
+          <List.Item.Meta
+            title={<Text strong>{item.title}</Text>}
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/lists/BasicList.jsx]: title's are already bold? Why are you adding `strong`?

File: src/main/webapp/resources/js/components/lists/BasicList.jsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import React from "react";
+import PropTypes from "prop-types";
+import { List, Typography } from "antd";
+
+const { Text } = Typography;
+
+export function BasicList({
+  itemLayout = "horizontal",
+  dataSource = { dataSource },
+  ...props
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/lists/BasicList.jsx]: Make sure you do something with these if you want them.

File: src/main/webapp/resources/js/components/lists/BasicList.jsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+          />
+        </List.Item>
+      )}
+    />
+  );
+}
+
+BasicList.propTypes = {
+  itemLayout: PropTypes.string,
+  dataSource: PropTypes.array.isRequired
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/lists/BasicList.jsx]: Can you look into `propTypes` for a list of objects that have a `title` and a `desc` attribute.

Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/lists/BasicList.jsx]: Added in 48b1a7b8bb8e39cb2a7eda2d963f27f207aae066

File: src/main/webapp/resources/js/components/lists/BasicList.jsx (hunk 5 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/lists/BasicList.jsx]: Unfortunately the titles aren't bolded by default :(

File: src/main/webapp/resources/js/components/lists/BasicList.jsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+const { Text } = Typography;
+
+export function BasicList({
+  itemLayout = "horizontal",
+  dataSource = { dataSource },
+  ...props
+}) {
+  return (
+    <List
+      itemLayout="horizontal"
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/lists/BasicList.jsx]: Fixed in 48b1a7b8bb8e39cb2a7eda2d963f27f207aae066

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx (hunk 4 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Added in 5e42840cf8b9aa72a4ab4384c86a925e103d0382

File: src/main/webapp/resources/js/components/tabs/SideTabs.jsx (hunk 3 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/tabs/SideTabs.jsx]: Fixed in 5e42840cf8b9aa72a4ab4384c86a925e103d0382 and 96a30eef5b4770f8733922e46814ef78f89cc36e

File: src/main/webapp/resources/js/components/tabs/SideTabs.jsx (hunk 2 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/tabs/SideTabs.jsx]: Fixed in 5e42840cf8b9aa72a4ab4384c86a925e103d0382

File: src/main/webapp/resources/js/components/alerts/WarningAlert.jsx (hunk 1 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/alerts/WarningAlert.jsx]: Done in 82cbf1d540c536326e9a74e802ccc1fd825abbf7

File: src/main/webapp/resources/js/components/lists/BasicList.jsx (hunk 6 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/lists/BasicList.jsx]: Removed in 48b1a7b8bb8e39cb2a7eda2d963f27f207aae066

File: src/main/webapp/resources/js/components/alerts/ErrorAlert.jsx [hunk 9]
```diff
@@ -1,19 +1,35 @@
...
 import { Alert } from "antd";
 
-export function ErrorAlert(props) {
+/**
+ * Stateless UI component for displaying an [antd error Alert]{@link https://ant.design/components/alert/}
+ *
+ * @param {string} message - Text to display in alert
+ * @param {string} description - Optional description
+ *
+ * @returns {antd Alert} - Returns an antd error 'Alert' component
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/alerts/ErrorAlert.jsx]: This is not a proper return type.  Take a look at JSDoc, but I think you can use `element`

File: src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx [hunk 10]
```diff
@@ -21,6 +21,7 @@ import { getI18N } from "../../../utilities/i18n-utilties";
 import { SPACE_SM } from "../../../styles/spacing";
 import { navigate } from "@reach/router";
 import { green6, grey6, red6 } from "../../../styles/colors";
+import { ContentLoading } from "../../../components/loader/ContentLoading";
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx]: In this file you also make"

```jsx
 <Icon type="check-circle"  style={{ marginRight: SPACE_SM, color: green6 }}  />
 <Icon type="close-circle" style={{ marginRight: SPACE_SM, color: red6 }} />
<Icon type="loading" style={{ marginRight: SPACE_SM, color: grey6 }} />
```
Can you make these standard components

Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx]: Done in 6de64b4b1c59f8f50d554093d2ced426d7b6d605

**Pull request merged**